.github/workflows/deploy.yml

```diff
@@ -0,0 +1,43 @@
+name: Deploy
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+
+jobs:
+  deploy:
+    name: Build and deploy
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to GHCR
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Create image tags
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ghcr.io/${{ github.repository }}
+          flavor: latest=true # adds :latest tag to outputs.tags
+          tags: type=sha,format=long,prefix= # adds :<sha> tag to outputs.tags
+
+      - name: Build and push
+        uses: docker/build-push-action@v6
+        with:
+          context: .
+          platforms: linux/amd64,linux/arm64/v8
+          push: true
```

Comment:
can we set this to true only on push to master? in this way, we avoid the permission error when coming from a fork

Reply:
Yes, I think we can:

- eac0a72 :  made the push dependent on push to master.
- 372bfc7: forgot to change the trigger back to master.

Reply:
I meant the the push should be conditional depending on the action trigger. True if push, false if pr

Reply:
Sorry I misunderstood. I fixed with [077150a](https://github.com/SciCatProject/oai-provider-service/pull/112/commits/077150ad9cf7dd90d847811fb1cac5c7c675cffa). Seems to work for 'push' on my branch and 'pr' on the scicat master.

Reply:
np! thanks for the changes! I would be fine merging now, do you plan any other change or is it good to go?

Reply:
I'm not planning any other changes in this PR so we are good to go from my side. Thanks a lot for your help.